### PR TITLE
Fix for loosing VMSS extension protected settings when adding nodes 

### PR DESCRIPTION
**Pull request by anuchandy**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by anuchandy** [comment]
Fixes https://github.com/Azure/azure-sdk-for-net/issues/2716

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute.Fluent/VirtualMachineScaleSetExtensionImpl.cs [hunk 1]
```diff
@@ -63,7 +60,16 @@ public bool AutoUpgradeMinorVersionEnabled()
...
         public IReadOnlyDictionary<string, object> PublicSettings()
         {
-            return this.publicSettings as IReadOnlyDictionary<string, object>;
+            if (this.Inner.Settings == null)
+            {
+                return new Dictionary<string, object>();
+            }
+            if (this.Inner.Settings is JObject)
+            {
+                var jObject = this.Inner.Settings as JObject;
```
Comment (hovsepm) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute.Fluent/VirtualMachineScaleSetExtensionImpl.cs]: >as JObject [](start = 50, length = 10)

Please change "as" cast to explicit conversion. there are to location in this method.

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute.Fluent/VirtualMachineScaleSetExtensionImpl.cs [hunk 2]
```diff
@@ -162,53 +181,52 @@ public VirtualMachineScaleSetExtensionImpl WithVersion(string extensionImageVers
...
+        private IDictionary<string, object> EnsurePublicSettings()
         {
-            if (Inner.Settings == null)
+            if (this.Inner.Settings == null)
             {
-                this.publicSettings = new Dictionary<string, object>();
-                Inner.Settings = this.publicSettings;
+                this.Inner.Settings = new Dictionary<string, object>();
+                return this.Inner.Settings as Dictionary<string, object>;
             }
```
Comment (hovsepm) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute.Fluent/VirtualMachineScaleSetExtensionImpl.cs]: do explicit cast please

**Review by hovsepm — CHANGES_REQUESTED** [review]
Please address review comments.

**Pull request merged**